calculate_imputation_factors_wrangler.py

```diff
@@ -0,0 +1,128 @@
+import json
+import traceback
+import random
+import os
+import pandas as pd
+import boto3
+
+
+def _get_traceback(exception):
+    """
+    Given an exception, returns the traceback as a string.
+
+    :param exception: Exception object
+    :return: string
+    """
+
+    return ''.join(
+        traceback.format_exception(
+            etype=type(exception), value=exception, tb=exception.__traceback__
+        )
+    )
+
+
+def lambda_handler(event, context):
+    """
+    Prepares data for and calls the Calculate imputation factors method.
+    - adds on the required columns needed by the method.
+
+    :param event: lambda event
+    :param context: lambda context
+    :return: string
+    """
+    # environment variables
+    sqs = boto3.client('sqs')
+    lambda_client = boto3.client('lambda')
+    checkpoint = os.environ['checkpoint']
+
+    queue_url = os.environ['queue_url']
+    sqs_messageid_name = os.environ['sqs_messageid_name']
+
+    current_period = os.environ['period']
+    method_name = os.environ['method_name']
+    questions = os.environ['questions']
+
+    try:
+        # read in data from the sqs queue
+        response = sqs.receive_message(QueueUrl=queue_url)
+        message = response['Messages'][0]
+        message_json = json.loads(message['Body'])
+
+        # reciept handler used to clear sqs queue
+        receipt_handle = message['ReceiptHandle']
+
+        data = pd.DataFrame(message_json)
+
+        # create df columns needed for method
+        for question in questions.split(' '):
+            data['imputation_factor_' + question] = 0
+
+        data_json = data.to_json(orient='records')
+
+        # invoke the method to calculate the factors
+        calculate_factors = lambda_client.invoke(FunctionName=method_name, Payload=data_json)
+        json_response = calculate_factors.get('Payload').read().decode("UTF-8")
+
+        # send data to sqs queue
+        send_sqs_message(queue_url, json.loads(json_response), sqs_messageid_name)
+
+        ### COMMENTED OUT FOR TESTING ###
+        # sqs.delete_message(QueueUrl=queue_url, ReceiptHandle=receipt_handle)
```

Comment:
This is commented out for testing. Should it be in or out?

Reply:
I have uncommented the code